app/controllers/transcripts_controller.rb

```diff
@@ -106,11 +106,12 @@ def set_transcript
       @transcript = TranscriptService.find_by_uid(params[:id])
     end
 
-    def set_transcript_if_admin
-      return if @transcript
-      return unless logged_in_user
-      return unless logged_in_user.admin? || logged_in_user.content_editor?
-      @transcript = Transcript.find_by(uid: params[:id])
+    def set_transcript_for_show
+      @transcript = if logged_in_user && (logged_in_user.admin? || logged_in_user.content_editor?)
```

Comment:
shall we move this logic to the service and make the controller simple ? Idea of services are to hold the logic and expose simple interfaces to controllers


E.g 

```
#service 

@transcript =  if logged_in_user && (logged_in_user.admin? || logged_in_user.content_editor?)
                             ..
                         else
                             ..
                         end

#controller 
@transcript = TranscriptService.find(params[:id])


```